qiskit/circuit/quantumcircuit.py

```diff
@@ -2565,8 +2565,19 @@ def add_calibration(self, gate, qubits, schedule, params=None):
         """
         if isinstance(gate, Gate):
             self._calibrations[gate.name][(tuple(qubits), tuple(gate.params))] = schedule
+            _name = gate.name
         else:
             self._calibrations[gate][(tuple(qubits), tuple(params or []))] = schedule
+            _name = gate
```

Comment:
If you still need `name` after fixing the issues below, I would suggest:
```suggestion
            name = gate.name
            self._calibrations[name][(tuple(qubits), tuple(gate.params))] = schedule
        else:
            name = gate
            self._calibrations[name][(tuple(qubits), tuple(params or []))] = schedule
```

Reply:
thanks for this, I'm committing a new version that incorporates your suggestion